File: api/src/main/java/net/blueshell/api/common/event/jpa/PrePersistEvent.java
package net.blueshell.api.common.event.jpa;

public class PrePersistEvent<T> extends BaseJpaEvent<T> {
    public PrePersistEvent(T source) {
        super(source);
    }

    public PrePersistEvent(T source,
                           java.util.Map<String, Object> beforeState,
                           java.util.Map<String, Object> afterState,
                           java.util.Map<String, ValueChange> changes) {
        super(source, beforeState, afterState, changes);
    }
}

File: api/src/main/kotlin/net/blueshell/api/common/event/jpa/PrePersistEvent.kt
package net.blueshell.api.common.event.jpa

import lombok.Getter
import org.springframework.core.ResolvableType
import org.springframework.core.ResolvableTypeProvider

class PrePersistEvent<T>(@field:Getter private val source: T?) : ResolvableTypeProvider {
    override fun getResolvableType(): ResolvableType? {
        return ResolvableType.forClassWithGenerics(
            javaClass, ResolvableType.forInstance(getSource())
        )
    }
}